src/features/navigator-interface.js

```diff
@@ -0,0 +1,20 @@
+import { defineProperty } from '../utils'
+
+export function init (args) {
+    try {
+        if (navigator.duckduckgo) return
+        defineProperty(Navigator.prototype, 'duckduckgo', {
+            value: {
+                platform: args.platform.name,
+                async isDuckDuckGo () {
+                    return true
+                }
+            },
+            enumerable: true,
```

Comment:
I verified this appears to behave the same as Brave where navigator.brave isn't writable but navigator.brave.isBrave is.